### PR TITLE
Run build on Python 3.8

### PR DESCRIPTION
**Pull request by lkollar**
Enable Travis and AppVeyor builds on 3.8.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -86,15 +95,15 @@ matrix:
...
+        # OSX 10.12 - xcode 8.3 - Python 3.8
         - os: osx
-          osx_image: xcode10.1
+          osx_image: xcode8.3
           compiler: clang
           env: >
             [
             CXX=clang++ ,
             CC=clang     ,
-            PYTHON_VERSION=3.7   ,
```
Comment (romainthomas) [inline .travis.yml]: I would not be ok to remove Python 3.7 from osx

Comment (lkollar) [reply .travis.yml]: Sorry, that was a mistake. Will revert.

**Comment by lkollar** [comment]
Looks like the windows build is blocked by https://github.com/appveyor/ci/issues/3142.

**Comment by romainthomas** [comment]
Yep. I'll wait for the support before merging. Thanks, though.

**Comment by hugovk** [comment]
🚀 Python 3.8 is now available on AppVeyor:

* https://github.com/appveyor/ci/issues/3142#issuecomment-552049896

**Comment by romainthomas** [comment]
Nice! Waiting for appveyor and I'll merge

**Comment by lkollar** [comment]
The Python tests are failing on Windows:

```
2474  C:\projects\lief\build\temp.win32-3.8\Release\tests\elf>C:/Python38/python.exe C:/projects/lief/tests/elf/elf_test.py 
2475CUSTOMBUILD : Fatal Python error : _PyInterpreterState_Get(): no current thread state [C:\projects\lief\build\temp.win32-3.8\Release\tests\check-lief.vcxproj]
2476  Python runtime state: initialized
2477  
2478  Current thread 0x00000418 (most recent call first):
2479    File "<frozen importlib._bootstrap>", line 219 in _call_with_frames_removed
2480    File "<frozen importlib._bootstrap_external>", line 1101 in create_module
2481    File "<frozen importlib._bootstrap>", line 556 in module_from_spec
2482    File "<frozen importlib._bootstrap>", line 657 in _load_unlocked
2483    File "<frozen importlib._bootstrap>", line 975 in _find_and_load_unlocked
2484    File "<frozen importlib._bootstrap>", line 991 in _find_and_load
2485    File "C:/projects/lief/tests/elf/elf_test.py", line 3 in <module>
2486  
2487          Start 182: ELF_PYTHON_add_section
2488
```
The two messages contradict each other: "_PyInterpreterState_Get(): no current thread state", then the fault handler reports "Python runtime state: initialized". This could happen if there were multiple copies of the interpreter state present.

On Python 3.8 extensions shouldn't be linked to `libpython` anymore. I'm not sure how this affects Windows but I wonder if it is possible that LIEF links with `libpython3.8` on Windows and this causes duplicate symbols (like the global thread state).

**Comment by lkollar** [comment]
I tried modifying the CMake configuration to avoid linking with libpython, but it didn't help. I don't use Windows so I'm pretty lost trying to debug this... Any tips are welcome. 

I'll be away for a week so I won't be able to work on this for a while.

**Comment by romainthomas** [comment]
Same for me, I don't work on windows. Btw, I started to update the version of Pybind11 (v2.4.3). I still need to fix some issues related to enums.

**Comment by romainthomas** [comment]
Commit 934d84ffea8ceeba0d89e13990d4d05a74c6b593 updates the Pybind11 version (cc @wisk)

**Comment by lkollar** [comment]
I rebased the branch on top of the latest master but the Windows build is still failing of Python 3.8.

**Comment by romainthomas** [comment]
I merged the the PR in commits 5f865a8980bbfdd6ec66cab2dad6649f7949095b and 42ee4173cd6545ed2532fc603b67e0e17a3b1a6d. I keep Python 3.8 disabled on AppVeyor while there is this error.

**Comment by lkollar** [comment]
Great, thanks @romainthomas!

**Pull request closed without merging**